### ISSUE TITLE
`include Dependency` breaks strict null object behavior

### ISSUE THREAD

**Issue by strathmeyer** [eventide-project/dependency#4]
```
class Bar
  def bar; end
end

class Baz
  include Dependency
  def baz; end
end

class Foo
  include Dependency
  dependency :bar, Bar
  dependency :baz, Baz
end
```

Expected:  The baz dependency would only respond to the baz method.

Actual: The baz dependency responds to all methods.

```
f = Foo.new
f.baz
# => #<Mimic::Class::Object_180:0x00007fa6a80f3da8>
f.baz.qux
# => nil
f.bar
# => #<Mimic::Class::Bar_220:0x00007fa6a807a340>
f.bar.bar
# => nil
f.bar.qux
# NoMethodError (undefined method `qux' for #<Mimic::Class::Bar_200:0x00007fa67c918938>)
```

The root cause is that doing `include Dependency` puts `SubstAttr::Substitute` into the hierarchy:

```
Baz.const_defined?(:Substitute, true)
# => true
Baz.const_get(:Substitute, true)
# => SubstAttr::Substitute
```

**Comment by sbellware** [comment]
@strathmeyer Yup, that sure looks like a legitimate mistake in the implementation. I'll take a look as soon as I have a chance. Thanks for finding the problem and reporting it!

**Comment by ntl** [comment]
Thanks for reporting, @strathmeyer. I've been able to reproduce the issue more directly. Here is a code snippet that works, because `SomeClass` does not include the `Dependency` module:

``` ruby
class SomeClass
end

some_class_substitute = Dependency::Substitute.build(SomeClass)

# Raises a NoMethodError, as expected
some_class_substitute.some_unknown_method
```

Here is a snippet of code that malfunctions, because `OtherClass` includes the `Dependency` module:

``` ruby
class OtherClass
  include Dependency
end

other_class_substitute = Dependency::Substitute.build(OtherClass)

# Does not raise a NoMethodError, which is a bug
other_class_substitute.some_unknown_method
```

**Comment by sbellware** [comment]
I haven't looked in detail, but I have a suspicion in principle that the protocol discovery is too naive - making use of the general case code, and ending up falling back on a permissive null object rather than a strict null object. Shouldn't be too much effort to fix. There'll be more effort in shaping new test cases and controls than implementing the correction.

**Comment by ntl** [comment]
This issue is fixed by v2.1.0.0 (in conjunction with v2.2.0.0 of `evt-subst_attr`). Sorry for the delay! It's been a busy summer.

For posterity, the malfunction was caused by the fact that Ruby's `include`, when used to include a module into a class, also imports the module's namespace into the target class's namespace. I love Ruby, but sure wish that "include this mixin into this class" and "import that namespace into this one" were separate things.
